### ISSUE TITLE
Rails 5 question: Unable to use `rails db:setup` to create Postgres database on OSX

### ISSUE THREAD

**Issue by SpacyRicochet** [rails/rails#23181]
I want to create a web API service that does location calculations. After some research I found that Postgres would a good database model for that, so I decided to create a Rails API app with Postgres.

After installing postgres with Homebrew and following the instructions to launch it (and add it to the LaunchAgents), I created a new Rails 5 API app and wanted to run `bundle exec rails db:setup`. However, this fails with the following message:

```
Couldn't create database for {"adapter"=>"postgresql", "encoding"=>"unicode", "pool"=>5, "database"=>"todoapi_development"}
rails aborted!
PG::ConnectionBad: could not connect to server: No such file or directory
	Is the server running locally and accepting
	connections on Unix domain socket "/tmp/.s.PGSQL.5432"?
```

Confusing for a very new Rails developer that hasn't touched Postgres before. After trying lots of Googl-ing for the answer, I still can't come up with a solution. 

* Most of the answers ask you to switch to the `postgres` user. Somehow this always asks me for password and I'm at a loss as to what that password should be. 
  * Often, sites mention 'The password doesn't matter. Postgres defaults to `ident` which is the server user.' and assume you just log in. No answers as to why it asks for an unknown password for me.

I'm pretty sure this has to do with some kind of initial step that I forgot to do and that isn't as well documented in the Homebrew installation, or maybe just dead obvious to other developers. 

But does anyone know how to get Postgres started in the Rails app?

**Comment by pixeltrix** [comment]
@SpacyRicochet I've just tried with Rails 5.0.0.beta1 and it created a database fine - are you sure the database is running? Does the socket file actually exist in the temp directory?

**Comment by SpacyRicochet** [comment]
> are you sure the database is running? Does the socket file actually exist in the temp directory?

This is where my new-ness comes through. Can you help me with checking for these?

```
➜  ~  launchctl list | grep sql
-	1	homebrew.mxcl.postgresql
```

So I _think_ Postgres is running?

And with socket file, I assume you mean `/tmp/.s.PGSQL.5432`? Doing `ls -a /tmp | grep PGSQL` gives me nothing. So no, the socket file doesn't exist.

The startup launch symlinks to the Homebrew postgres plists are installed. And I have launched it with launchctl already as well.

**Comment by SpacyRicochet** [comment]
Solved! 

*Solution, short version:* The original database in `/usr/local/var/postgres` was created with Postgres 9.4 and homebrew installs 9.5. This causes the server to crap out. I deleted the database directory, reinstalled postgres and this issue seems solved.

*How this happened:* I started preparation for this project in december and followed an 'install ruby, rails and postgres' guide. This probably installed 9.4 and created the database. Since 9.5 is a very recent release (January 7th), that new database already wasn't compatible.

@pixeltrix Thanks for thinking along :)
